lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -4,68 +4,45 @@ class Deployment < KubernetesResource
     TIMEOUT = 5.minutes
 
     def sync
-      json_data, _err, st = kubectl.run("get", type, @name, "--output=json")
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
       @found = st.success?
-      @rollout_data = {}
-      @status = nil
-      @representative_pod = nil
-      @pods = []
 
       if @found
-        @rollout_data = JSON.parse(json_data)["status"]
-          .slice("updatedReplicas", "replicas", "availableReplicas", "unavailableReplicas")
-        @status = @rollout_data.map { |replica_state, num| "#{num} #{replica_state}" }.join(", ")
-
-        pod_list, _err, st = kubectl.run("get", "pods", "-a", "-l", "name=#{name}", "--output=json")
-        if st.success?
-          pods_json = JSON.parse(pod_list)["items"]
-          pods_json.each do |pod_json|
-            pod_name = pod_json["metadata"]["name"]
-            pod = Pod.new(
-              name: pod_name,
-              namespace: namespace,
-              context: context,
-              file: nil,
-              parent: "#{@name.capitalize} deployment",
-              logger: @logger
-            )
-            pod.deploy_started = @deploy_started
-            pod.interpret_json_data(pod_json)
-
-            if !@representative_pod && pod_probably_new?(pod_json)
-              @representative_pod = pod
-            end
-            @pods << pod
-          end
-        end
+        deployment_data = JSON.parse(raw_json)
+        @latest_rs = find_latest_rs(deployment_data)
+        @rollout_data = { "replicas" => 0 }.merge!(deployment_data["status"]
```

Comment:
no good reason. I refactored this section a bunch of times, so maybe there was some intermediate version that didn't use the return value. I'll fix it.